### ISSUE TITLE
AnySplat using original input image sizes

### ISSUE THREAD

**Issue by dsvilarkovic** [OpenRobotLab/AnySplat#66]
Hello, 
I am aware that your codebase works best when running 448x448 input with center cropping, but was wondering is there a way I can still run AnySplat on original input images and get the good results. I am trying to do this because I don't have a good coverage with initial input images when using central cropping. 

For my input view of 10 images I have feedforwarded to 3DGS and then rendered with original intrinsics/extrinsics I get this video: 

https://github.com/user-attachments/assets/ed1d4321-194a-4759-9611-8826addc28ca

Now, I would like to be able to have white regions filled in with original pixels somehow. This comes from input cropping missing a lot of sideview pixels.

I have created my wrapper around AnySplat implementation and the function for running code is this:

```python
    
    #  model = AnySplat.from_pretrained(model_path)
    #  self.model = model
    #  self.patch_size = self.model.encoder.aggregator.patch_size


    def inference_tensor(
            self,
            images_bhwc: torch.Tensor,
            downscale_and_crop_image_for_anysplat = True,
    ) -> Tuple[EncoderOutput, DecoderOutput, torch.Tensor, Tuple[int, int], Tuple[int, int]]:
        """
        Run inference on a batch of images.

        Args:
            images_bhwc (torch.Tensor): Tensor of shape [V, H, W, C], 
                with values in the [0, 1] range.
            downscale_and_crop_image_for_anysplat (bool): Whether to downscale and center crop
                the input images to 448x448, as required by AnySplat.

        Returns:
            Tuple containing:
                - EncoderOutput: The output object from the encoder, 
                see `anysplat.model.encoder`.
                - DecoderOutput: The output object from the decoder, 
                see `anysplat.model.decoder`.
                - torch.Tensor: Re-formatted image tensor of shape [V, C, H, W], 
                in [0, 1] range before mapping.
                - Tuple[int, int]: Original image spatial size (height, width).
                - Tuple[int, int]: Padded image spatial size (height, width).
        """

        b, height, width, _ = images_bhwc.shape

        logger.info(f"Input images shape: {images_bhwc.shape}, dtype: {images_bhwc.dtype}, device: {images_bhwc.device}")
        logger.info(f"Using downscale_and_crop_image_for_anysplat = {downscale_and_crop_image_for_anysplat}")

        if downscale_and_crop_image_for_anysplat:
            if width > height:
                new_height = 448
                new_width = int(width * (new_height / height))
            else:
                new_width = 448
                new_height = int(height * (new_width / width))

            new_image_shape_hw = (new_height, new_width)
            old_image_shape_hw = (height, width)

        else:
            new_height = int(math.ceil(height / self.patch_size) * self.patch_size)
            new_width = int(math.ceil(width / self.patch_size) * self.patch_size)

            new_image_shape_hw = (new_height, new_width)
            old_image_shape_hw = (height, width)

        logger.info(f"Resizing input images to {new_image_shape_hw} from original {old_image_shape_hw}...")
        if downscale_and_crop_image_for_anysplat:
            images_bchw_tensor = [process_image(img) for img in images_bhwc.squeeze(0)]
            images_bchw_tensor = torch.stack(images_bchw_tensor, dim=0).to(self.device)
        else:
            # Do resize (instead of padding)
            images_bchw_tensor = images_bhwc.permute(0, 3, 1, 2).contiguous()  # [V, C, H, W], values in [0, 1]

            # Resize to the target multiples of patch size (non-uniform scale OK)
            images_bchw_tensor = torch.nn.functional.interpolate(
                images_bchw_tensor,
                size=(new_height, new_width),
                mode="bilinear",
                align_corners=False,
                antialias=True,  
            )

            images_bchw_tensor = images_bchw_tensor.to(self.device) * 2.0 - 1.0  # to [-1, 1]


        with torch.no_grad():
            encoder_output, decoder_output = self.model((images_bchw_tensor.unsqueeze(0) + 1) * 0.5)

        return encoder_output, decoder_output, images_bchw_tensor, new_image_shape_hw, old_image_shape_hw

```

But I still get bad results when using original image inputs (resized to match patch size of 14):

https://github.com/user-attachments/assets/cd94ee67-8eae-4200-b57b-ffce4badfc4d

Am I doing something obviously wrong, or there are some tricks or setups I can use to run the model in non-448x448 shape?

**Comment by myc634** [comment]
Hi, I think the main reason lies in your input view selection strategy, keeping a suitable overlap between the input images is important for getting a good result, another advise is that Anysplat strongly relies on VGGT, so if you find a case which fails on VGGT, it might not also get a good result on AnySplat

**Comment by peizhou99** [comment]
> Hello, I am aware that your codebase works best when running 448x448 input with center cropping, but was wondering is there a way I can still run AnySplat on original input images and get the good results. I am trying to do this because I don't have a good coverage with initial input images when using central cropping.
> 
> For my input view of 10 images I have feedforwarded to 3DGS and then rendered with original intrinsics/extrinsics I get this video:
> 
>  anysplat_input_downscale_and_crop_image_for_anysplat_True.mp4 
> Now, I would like to be able to have white regions filled in with original pixels somehow. This comes from input cropping missing a lot of sideview pixels.
> 
> I have created my wrapper around AnySplat implementation and the function for running code is this:
> 
>     
>     #  model = AnySplat.from_pretrained(model_path)
>     #  self.model = model
>     #  self.patch_size = self.model.encoder.aggregator.patch_size
> 
> 
>     def inference_tensor(
>             self,
>             images_bhwc: torch.Tensor,
>             downscale_and_crop_image_for_anysplat = True,
>     ) -> Tuple[EncoderOutput, DecoderOutput, torch.Tensor, Tuple[int, int], Tuple[int, int]]:
>         """
>         Run inference on a batch of images.
> 
>         Args:
>             images_bhwc (torch.Tensor): Tensor of shape [V, H, W, C], 
>                 with values in the [0, 1] range.
>             downscale_and_crop_image_for_anysplat (bool): Whether to downscale and center crop
>                 the input images to 448x448, as required by AnySplat.
> 
>         Returns:
>             Tuple containing:
>                 - EncoderOutput: The output object from the encoder, 
>                 see `anysplat.model.encoder`.
>                 - DecoderOutput: The output object from the decoder, 
>                 see `anysplat.model.decoder`.
>                 - torch.Tensor: Re-formatted image tensor of shape [V, C, H, W], 
>                 in [0, 1] range before mapping.
>                 - Tuple[int, int]: Original image spatial size (height, width).
>                 - Tuple[int, int]: Padded image spatial size (height, width).
>         """
> 
>         b, height, width, _ = images_bhwc.shape
> 
>         logger.info(f"Input images shape: {images_bhwc.shape}, dtype: {images_bhwc.dtype}, device: {images_bhwc.device}")
>         logger.info(f"Using downscale_and_crop_image_for_anysplat = {downscale_and_crop_image_for_anysplat}")
> 
>         if downscale_and_crop_image_for_anysplat:
>             if width > height:
>                 new_height = 448
>                 new_width = int(width * (new_height / height))
>             else:
>                 new_width = 448
>                 new_height = int(height * (new_width / width))
> 
>             new_image_shape_hw = (new_height, new_width)
>             old_image_shape_hw = (height, width)
> 
>         else:
>             new_height = int(math.ceil(height / self.patch_size) * self.patch_size)
>             new_width = int(math.ceil(width / self.patch_size) * self.patch_size)
> 
>             new_image_shape_hw = (new_height, new_width)
>             old_image_shape_hw = (height, width)
> 
>         logger.info(f"Resizing input images to {new_image_shape_hw} from original {old_image_shape_hw}...")
>         if downscale_and_crop_image_for_anysplat:
>             images_bchw_tensor = [process_image(img) for img in images_bhwc.squeeze(0)]
>             images_bchw_tensor = torch.stack(images_bchw_tensor, dim=0).to(self.device)
>         else:
>             # Do resize (instead of padding)
>             images_bchw_tensor = images_bhwc.permute(0, 3, 1, 2).contiguous()  # [V, C, H, W], values in [0, 1]
> 
>             # Resize to the target multiples of patch size (non-uniform scale OK)
>             images_bchw_tensor = torch.nn.functional.interpolate(
>                 images_bchw_tensor,
>                 size=(new_height, new_width),
>                 mode="bilinear",
>                 align_corners=False,
>                 antialias=True,  
>             )
> 
>             images_bchw_tensor = images_bchw_tensor.to(self.device) * 2.0 - 1.0  # to [-1, 1]
> 
> 
>         with torch.no_grad():
>             encoder_output, decoder_output = self.model((images_bchw_tensor.unsqueeze(0) + 1) * 0.5)
> 
>         return encoder_output, decoder_output, images_bchw_tensor, new_image_shape_hw, old_image_shape_hw
> But I still get bad results when using original image inputs (resized to match patch size of 14):
> 
>  anysplat_input_downscale_and_crop_image_for_anysplat_False.mp4 
> Am I doing something obviously wrong, or there are some tricks or setups I can use to run the model in non-448x448 shape?

I found the same issue, too. And I don't know how to solve it
